src/Aspire.Hosting/Eventing/DistributedApplicationEventing.cs

```diff
@@ -25,26 +25,38 @@ public async Task PublishAsync<T>(T @event, EventDispatchBehavior dispatchBehavi
     {
         if (_eventSubscriptionListLookup.TryGetValue(typeof(T), out var subscriptions))
         {
+            // Determine the resource associated with the event if it's a resource-specific event
+            var resource = @event is IDistributedApplicationResourceEvent resourceEvent ? resourceEvent.Resource : null;
+
             if (dispatchBehavior == EventDispatchBehavior.BlockingConcurrent || dispatchBehavior == EventDispatchBehavior.NonBlockingConcurrent)
             {
                 var pendingSubscriptionCallbacks = new List<Task>(subscriptions.Count);
                 foreach (var subscription in subscriptions.ToArray())
                 {
-                    var pendingSubscriptionCallback = subscription.Callback(@event, cancellationToken);
-                    pendingSubscriptionCallbacks.Add(pendingSubscriptionCallback);
+                    // Wrap each callback to catch exceptions individually
+                    var wrappedCallback = Task.Run(async () =>
+                    {
+                        try
+                        {
+                            await subscription.Callback(@event, cancellationToken).ConfigureAwait(false);
+                        }
+                        catch (Exception ex)
+                        {
+                            await PublishExceptionEventAsync(ex, typeof(T), resource).ConfigureAwait(false);
+                            throw;
+                        }
+                    }, cancellationToken);
+                    pendingSubscriptionCallbacks.Add(wrappedCallback);
                 }
 
                 if (dispatchBehavior == EventDispatchBehavior.NonBlockingConcurrent)
                 {
-                    // Non-blocking concurrent.
-                    _ = Task.Run(async () =>
-                    {
-                        await Task.WhenAll(pendingSubscriptionCallbacks).ConfigureAwait(false);
-                    }, default);
+                    // Non-blocking concurrent - fire and forget
+                    _ = Task.WhenAll(pendingSubscriptionCallbacks);
```

Comment:
This may not be ideal. You don't necessarily get to choose who your fire and forget partners are. If there is an event that is going to hang around waiting for interaction (for example) it would stop you being notified that your event threw an exception.